tests/baselines/reference/mappedTypeWithAsClauseAndLateBoundProperty2.js

```diff
@@ -9,101 +9,4 @@ export const thing = null;
-    reduce: {
-        (callbackfn: (previousValue: number, currentValue: number, currentIndex: number, array: number[]) => number): number;
-        (callbackfn: (previousValue: number, currentValue: number, currentIndex: number, array: number[]) => number, initialValue: number): number;
-        <U>(callbackfn: (previousValue: U, currentValue: number, currentIndex: number, array: number[]) => U, initialValue: U): U;
-    };
-    reduceRight: {
-        (callbackfn: (previousValue: number, currentValue: number, currentIndex: number, array: number[]) => number): number;
-        (callbackfn: (previousValue: number, currentValue: number, currentIndex: number, array: number[]) => number, initialValue: number): number;
-        <U>(callbackfn: (previousValue: U, currentValue: number, currentIndex: number, array: number[]) => U, initialValue: U): U;
-    };
-    find: {
-        <S extends number>(predicate: (value: number, index: number, obj: number[]) => value is S, thisArg?: any): S;
-        (predicate: (value: number, index: number, obj: number[]) => unknown, thisArg?: any): number;
-    };
-    findIndex: (predicate: (value: number, index: number, obj: number[]) => unknown, thisArg?: any) => number;
-    fill: (value: number, start?: number, end?: number) => number[];
-    copyWithin: (target: number, start: number, end?: number) => number[];
-    entries: () => ArrayIterator<[number, number]>;
-    keys: () => ArrayIterator<number>;
-    values: () => ArrayIterator<number>;
-    includes: (searchElement: number, fromIndex?: number) => boolean;
-    flatMap: <U, This = undefined>(callback: (this: This, value: number, index: number, array: number[]) => U | readonly U[], thisArg?: This) => U[];
-    flat: <A, D extends number = 1>(this: A, depth?: D) => FlatArray<A, D>[];
-    [Symbol.iterator]: () => ArrayIterator<number>;
-    readonly [Symbol.unscopables]: {
-        [x: number]: boolean;
-        length?: boolean;
-        toString?: boolean;
-        toLocaleString?: boolean;
-        pop?: boolean;
-        push?: boolean;
-        concat?: boolean;
-        join?: boolean;
-        reverse?: boolean;
-        shift?: boolean;
-        slice?: boolean;
-        sort?: boolean;
-        splice?: boolean;
-        unshift?: boolean;
-        indexOf?: boolean;
-        lastIndexOf?: boolean;
-        every?: boolean;
-        some?: boolean;
-        forEach?: boolean;
-        map?: boolean;
-        filter?: boolean;
-        reduce?: boolean;
-        reduceRight?: boolean;
-        find?: boolean;
-        findIndex?: boolean;
-        fill?: boolean;
-        copyWithin?: boolean;
-        entries?: boolean;
-        keys?: boolean;
-        values?: boolean;
-        includes?: boolean;
-        flatMap?: boolean;
-        flat?: boolean;
-        [Symbol.iterator]?: boolean;
-        readonly [Symbol.unscopables]?: boolean;
-    };
-};
+export declare const thing: { [K in keyof number[] as Exclude<K, "length">]: (number[])[K]; };

```

Comment:
There is a distinct possibility that improvements to node reuse in this PR have made this test not test what was intended anymore (knowing if the late bound property is included in the remapped type). If we can bail out of the node reuse use via an indirection, that'd probably be a good thing.

Reply:
Added `satisfies any` which the syntactic printer will not look past